subsys/bluetooth/audio/bap_unicast_server.c

```diff
@@ -28,6 +29,21 @@ LOG_MODULE_REGISTER(bt_bap_unicast_server, CONFIG_BT_BAP_UNICAST_SERVER_LOG_LEVE
 
 static const struct bt_bap_unicast_server_cb *unicast_server_cb;
 
+int bt_bap_unicast_server_register(const struct bt_bap_unicast_server_register_param *param)
+{
+	if (param == NULL) {
+		LOG_DBG("param is NULL");
+		return -EINVAL;
+	}
+
+	return bt_ascs_register(param->snk_cnt, param->src_cnt);
+}
+
+int bt_bap_unicast_server_unregister(void)
+{
+	return bt_ascs_unregister();
+}
+
 int bt_bap_unicast_server_register_cb(const struct bt_bap_unicast_server_cb *cb)
```

Comment:
Since this now shall be called after `bt_bap_unicast_server_register`, suggest to add a check and test for it

Reply:
This is already checked and tested, but it happens in ascs, through the global struct ascs and its member registered.

Reply:
I don't see a test nor any code that verifies that `bt_bap_unicast_server_register` is called before `bt_bap_unicast_server_register_cb`. 

Can you point me to it?

Reply:
function 'test_ascs_register_twice' in 'zephyr/tests/bluetooth/audio/ascs/src/main.c'

Reply:
For some reason I was thinking about register, not unregister. You are correct, Ill add this test case